### DISCUSSION TITLE
H1 LAN work mode investigation

### DISCUSSION THREAD

**Discussion by canton7** [nathanmarlor/foxess_modbus#519] [General]
Hey @art103,

Would you mind carrying out a couple of tests for me, see if we can get Work Mode and similar working on the H1 over LAN?

If you're comfortable using [modpoll](https://www.modbusdriver.com/modpoll.html), could you please try the following:

```
modpoll.exe -m tcp -p 502 -a 247 -0 -1 -t 3 -c 1 -r 41000 <inverter IP address>
modpoll.exe -m tcp -p 502 -a 247 -0 -1 -t 4 -c 1 -r 41000 <inverter IP address>
modpoll.exe -m tcp -p 502 -a 247 -0 -1 -t 3 -c 2 -r 41007 <inverter IP address>
modpoll.exe -m tcp -p 502 -a 247 -0 -1 -t 4 -c 2 -r 41007 <inverter IP address>
```

Thanks!

**Comment by art103** [comment]
Unfortunately they all return "Illegal data address" for Holding and Input Read, and for Holding Write.
I've thought about this before and wonder if the settings are relocated in the 31000 range somewhere?

Just to reassure you it works for supported registers: 😄 

```
$ mbpoll 10.x.x.x -a247 -t 4 -0 -1 -r 31027
mbpoll 1.4-25 - FieldTalk(tm) Modbus(R) Master Simulator
Copyright © 2015-2019 Pascal JEAN, https://github.com/epsilonrt/mbpoll
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; type 'mbpoll -w' for details.

Protocol configuration: Modbus TCP
Slave configuration...: address = [247]
                        start reference = 31027, count = 1
Communication.........: 10.x.x.x, port 502, t/o 1.00 s, poll rate 1000 ms
Data type.............: 16-bit register, output (holding) register table

-- Polling slave 247...
[31027]: 	2
```

**Reply by canton7** [reply]
Boo that's a shame, thanks! So the H1 over LAN is the only inverter configuration where we can't write the Work Mode currently.

**Reply by art103** [reply]
It's an inconvenience, but not the end of the world. It makes it slightly harder for less HW inclined folks to fully integrate, but is still great for monitoring.
FoxESS have removed the LAN port on the AC1-5.0-E I got recently, and already [fed back to someone else](https://www.eevblog.com/forum/dodgy-technology/foxess-inverter-ethernet-data-to-pc-data-and-other-monitoring-lie/) that it is not supported.
Looks a TCP / RTU ModBus bridge will be needed for full functionality 😄.

**Reply by bartashevich** [reply]
Hey. on new firmware 1.71 ethernet support is back. Just tested and it is working. At least for H1 inverter.
